src/backend/distributed/utils/maintenanced.c

```diff
@@ -146,6 +147,12 @@ InitializeMaintenanceDaemon(void)
 void
 InitializeMaintenanceDaemonForAdminDB(void)
 {
+	if (strcmp(ControlDbName, "") == 0)
```

Comment:
Are GUCs already read in _PG_init? If not, this would always return true.

Reply:
Yes, it appears that they are read since this change is able to start a maintenance deamon in the controldb even if there is no connection.

It is also possible to move this check to CitusMaintenanceDaemonMain, Then a background worker will start but then quit immediately.